src/transformers/models/jamba/modeling_jamba.py

```diff
@@ -0,0 +1,2127 @@
+            cache shape[2] is a dummy "seqlen" dimension to match the number of attention cache dimensions. For mamba,
+            the cache doesn't grow with seqlen so this dimension is always 1.
+
+            Contains pre-computed hidden-states (key and values in the self-attention blocks and convolution and
+            ssm states in the mamba blocks) that can be used (see `past_key_values` input) to speed up sequential decoding.
+
+            If `past_key_values` are used, the user can optionally input only the last `input_ids` (those that
+            don't have their past key value states given to this model) of shape `(batch_size, 1)` instead of all
+            `input_ids` of shape `(batch_size, sequence_length)`.
+        inputs_embeds (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`, *optional*):
+            Optionally, instead of passing `input_ids` you can choose to directly pass an embedded representation. This
+            is useful if you want more control over how to convert `input_ids` indices into associated vectors than the
+            model's internal embedding lookup matrix.
+        use_cache (`bool`, *optional*):
+            If set to `True`, `past_key_values` key value states are returned and can be used to speed up decoding (see
+            `past_key_values`).
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        output_router_logits (`bool`, *optional*):
+            Whether or not to return the logits of all the routers. They are useful for computing the router loss, and
+            should not be returned during inference.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    "The bare Jamba Model outputting raw hidden-states without any specific head on top.",
+    JAMBA_START_DOCSTRING,
+)
+# Adapted from transformers.models.mistral.modeling_mistral.MistralModel with MISTRAL->JAMBA, Mistral->Jamba
+class JambaModel(JambaPreTrainedModel):
+    """
+    Transformer decoder consisting of *config.num_hidden_layers* layers. Each layer is a [`JambaDecoderLayer`]
+
+    Args:
+        config: JambaConfig
+    """
+
+    def __init__(self, config: JambaConfig):
+        super().__init__(config)
+        self.padding_idx = config.pad_token_id
+        self.vocab_size = config.vocab_size
+
+        # The maximum sequence length that this model is intended to be used with. Doesn't have any real effect
+        self.n_ctx = config.n_ctx
+
```

Comment:
```suggestion
```
unused

Reply:
See [this](https://github.com/huggingface/transformers/pull/29943#discussion_r1545580904) comment

Reply:
from the comment it does not seem need in the `JambaModel` and only in the `config` (Llama does not have it in the LlamaModel) am I right?

Reply:
Yes but then the automatic quality checks didn't like it because it was a config attribute that wasn't used anywhere.. `modeling_llama.py` does use `max_position_embeddings` in some places in the code..

Reply:
The check can be fixed! with a variable like `SPECIAL_CASES_TO_ALLOW`

Reply:
https://github.com/huggingface/transformers/blob/416711c3ea88109cf25a9c5f85b4aeee2cb831b5/utils/check_config_attributes.py#L34

Reply:
Done